MyTasks/Task2_1/UserAndBotDialogue.cs

```diff
@@ -0,0 +1,60 @@
+﻿
+Console.WriteLine("Здравствуйте! Введите Ваше имя:");
+var userName = Console.ReadLine();
+
+while (char.IsLower(userName, 0))
+{
+    Console.WriteLine("Пожалуйста, введите имя с заглавной буквы:");    // "обработка" неправильного пользовательского ввода
+    userName = Console.ReadLine();
+}
+
+foreach (var symbol in userName)   // где-то в этом блоке баг: если первый ввод некорректный, то идет
+                                    // бесконечный вывод "Имя должно состоять из букв" в ответ
+                                    // как на последующий некорректный ввод, так и на корректный.
+{
+    while (char.IsNumber(symbol) || char.IsPunctuation(symbol) || char.IsWhiteSpace(symbol))
+    {
+        Console.WriteLine("Имя должно состоять из букв.");
+        userName = Console.ReadLine();
+    }
+}
+
+Console.WriteLine("Введите Вашу фамилию:");
+var userSurname = Console.ReadLine();
+
+while (char.IsLower(userSurname, 0))
+{
+    Console.WriteLine("Пожалуйста, введите фамилию с заглавной буквы:");    // "обработка" неправильного пользовательского ввода
+    userSurname = Console.ReadLine();
+}
+
+foreach (var symbol in userSurname)    //где-то в этом блоке баг: смотри строки 10-12
+{
+    while (char.IsNumber(symbol) || char.IsPunctuation(symbol) || char.IsWhiteSpace(symbol))
+    {
+        Console.WriteLine("Фамилия должна состоять из букв.");
+        userSurname = Console.ReadLine();
+    }
+}
+
+Console.WriteLine("Введите дату приема в формате дд/мм/гггг:");
+var date = new DateTime();
+date = DateTime.Parse(Console.ReadLine());
+
+var currentDate = new DateTime();
+currentDate = DateTime.Today;
+var comparisonResult = DateTime.Compare(date, currentDate); // Проверка ввода даты приема.
+                                                            // Если пользователь ввел дату из прошлого, то выводится сообщение
+                                                            // с просьбой ввести дату повторно.
+while (comparisonResult < 0)
+{
+    Console.WriteLine("Пожалуйста, введите актуальную дату:");
+    date = DateTime.Parse(Console.ReadLine());
+    break;
+}
```

Comment:
One time you entered wrong date - okey, second time it will not work and will return wrong time. 
You have var comparisonResult with comparing after you are setting date = DateTime.Parse(Console.ReadLine()) - so it will work once date = DateTime.Parse(Console.ReadLine()) in while loop - local variable, so it will not affect for your comparisonResult

Reply:
Done.